core/dbt/contracts/project.py

```diff
@@ -42,13 +42,19 @@ class Quoting(dbtClassMixin, Mergeable):
 
 @dataclass
 class Package(dbtClassMixin):
-    pass
+
+    def __post_serialize__(self, data, context: Optional[Dict]):
```

Comment:
The intent is to avoid changing the `sha1_hash` in `package-lock.yml` just because there's now an optional `name` attribute in the contract:
```
{'git': 'https://github.com/dbt-labs/dbt-utils', 'revision': None, 'warn-unpinned': None, 'subdirectory': None, 'unrendered': {'git': 'https://github.com/dbt-labs/dbt-utils'}, 'name': None}
```

But `name` actually _is_ a required field for `TarballPackage`

Reply:
This would be great to leave as a comment for the method.

Reply:
I can definitely add a comment. I'm also open to moving this logic elsewhere - perhaps [`_create_sha1_hash`](https://github.com/dbt-labs/dbt-core/blob/7cca8470e028221c583bbe4995663aa86d2b308a/core/dbt/task/deps.py#L42)?